controllers/activemqartemis_reconciler.go

```diff
@@ -2070,8 +2073,13 @@ func brokerPropertiesConfigSystemPropValue(customResource *brokerv1beta1.ActiveM
 
 	for _, extraSecretName := range customResource.Spec.DeploymentPlan.ExtraMounts.Secrets {
 		if strings.HasSuffix(extraSecretName, brokerPropsSuffix) {
-			// formated append to result with comma
-			result = fmt.Sprintf("%s,%s%s/", result, secretPathBase, extraSecretName)
+			if len(secretBrokerPropertiesData[extraSecretName]) == 1 {
+				// formated append to result with comma
+				result = fmt.Sprintf("%s,%s%s/", result, secretPathBase, extraSecretName)
+			} else {
+				// append to ordinal path
+				result = fmt.Sprintf("%s,%s%s/,%s%s/%s${STATEFUL_SET_ORDINAL}/", result, secretPathBase, extraSecretName, secretPathBase, extraSecretName, OrdinalPrefix)
+			}
```

Comment:
Secrets created by a customer could also include no broker.properties files. The check is not required, the ordinal path could be always appended because the broker skips not existing dirs, see https://github.com/apache/activemq-artemis/blob/2.33.0/artemis-server/src/main/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImpl.java#L551
```suggestion
			// append to ordinal path
			result = fmt.Sprintf("%s,%s%s/,%s%s/%s${STATEFUL_SET_ORDINAL}/", result, secretPathBase, extraSecretName, secretPathBase, extraSecretName, OrdinalPrefix)
```

Reply:
+1